core/src/main/java/com/radixdlt/rev2/REv2StateComputer.java

```diff
@@ -97,32 +98,46 @@ public final class REv2StateComputer implements StateComputerLedger.StateCompute
   private static final Logger log = LogManager.getLogger();
 
   private final RustStateComputer stateComputer;
+
+  // Maximum number of transactions to include in a proposal
   private final int maxNumTransactionsPerProposal;
+  // Maximum number of transaction payload bytes to include in a proposal
   private final int maxProposalTotalTxnsPayloadSize;
+  // Maximum number of transaction payload bytes to include in a proposal and its previous vertices
+  // chain.
+  // Effectively limits the size of a commit batch (i.e. the size of transactions under a single
+  // proof).
+  private final int maxProposalAndUncommittedVerticesTotalTxnPayloadSize;
   private final EventDispatcher<LedgerUpdate> ledgerUpdateEventDispatcher;
 
   private final EventDispatcher<MempoolAddSuccess> mempoolAddSuccessEventDispatcher;
   private final EventDispatcher<ConsensusByzantineEvent> consensusByzantineEventEventDispatcher;
   private final Serialization serialization;
   private final Hasher hasher;
+  private final Metrics metrics;
 
   public REv2StateComputer(
       RustStateComputer stateComputer,
       int maxNumTransactionsPerProposal,
       int maxProposalTotalTxnsPayloadSize,
+      int maxProposalAndUncommittedVerticesTotalTxnPayloadSize,
       Hasher hasher,
       EventDispatcher<LedgerUpdate> ledgerUpdateEventDispatcher,
       EventDispatcher<MempoolAddSuccess> mempoolAddSuccessEventDispatcher,
       EventDispatcher<ConsensusByzantineEvent> consensusByzantineEventEventDispatcher,
-      Serialization serialization) {
+      Serialization serialization,
+      Metrics metrics) {
     this.stateComputer = stateComputer;
     this.maxNumTransactionsPerProposal = maxNumTransactionsPerProposal;
     this.maxProposalTotalTxnsPayloadSize = maxProposalTotalTxnsPayloadSize;
+    this.maxProposalAndUncommittedVerticesTotalTxnPayloadSize =
```

Comment:
Ahh so this is really `maxUncommittedUserTransactionsTotalTxnPayloadSize`? (ie Transactions as opposed to Vertices)

I think it'd make sense to have `maxUncommittedUserTransactionsTotalCount` - and I guess to validate this during proposal validation time, based on the "Committed" in its parent QC.

But these can all wait for when we implement that ticket in the work tracker 👍

Reply:
> Ahh so this is really maxUncommittedUserTransactionsTotalTxnPayloadSize? (ie Transactions as opposed to Vertices)

yeah, I guess so :sweat_smile:
naming is hard sometimes, what I meant was
`maxProposalAndUncommittedVerticesTotalTxnPayloadSize` == `max` `TotalTxnPayloadSize` in `Proposal` `And` `UncommittedVertices`

Reply:
going for `maxUncommittedUserTransactionsTotalPayloadSize` then :+1:  (without the redundant `Txn`)